ntoskrnl/ke/i386/zeropage.S

```diff
@@ -0,0 +1,49 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Fast zeroing of pages
+ * COPYRIGHT:   Copyright 2021 Jérôme Gardou <jerome.gardou@reactos.org>
+ */
+
+#include <asm.inc>
+
+/* FUNCTIONS ****************************************************************/
+.code
+
+KeZeroPagesNoSSE:
+    push edi
+    mov edi, ecx
+    mov ecx, edx
+    shr ecx, 2
+    xor eax, eax
+    rep stosd
+    pop edi
+    ret
+
+PUBLIC @KeZeroPages@8
+@KeZeroPages@8:
+    /* Kernel will overwrite this with nop if SSE is available */
+    jmp KeZeroPagesNoSSE
+    /* Save XMM0 & zero it */
+    sub esp, 31
```

Comment:
I know x86 doesn't have a stack alignment requirement, but this seems like it would throw off pretty much any other code looking at this stack pointer (e.g. the aforementioned interrupt). Do you think it's safe?

Reply:
Indeed. I don't even know why I just dont sub 32 there...